client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -183,43 +145,55 @@ def get_curl(self):
         """
         headers = self.headers()
         headers.update({"Content-Type": "application/json"})
-        curl = "curl -X POST {} -d '' -H \"Content-Type: application/json\"".format(
-            self.prediction_url
-        )
+        curl = f"curl -X POST {self.prediction_url} -d '' -H \"Content-Type: application/json\""
         for header, value in headers.items():
-            curl += ' -H "{}: {}" '.format(header, value)
+            curl += f" -H \"{header}: {value}\""
         return curl
 
+    # TODO: Removed deprecated `always_retry_404` and `always_retry_429` params
     def predict(
         self,
-        x,
+        x: List[Any],
         compress=False,
-        max_retries=5,
-        always_retry_404=True,
-        always_retry_429=True,
-    ):
+        prediction_id: Optional[str] = None,
+        max_retries: Optional[int] = http_session_utils.DEFAULT_MAX_RETRIES,
+        always_retry_404=False,
+        always_retry_429=False,
+        retry_status: Optional[Set[int]] = http_session_utils.DEFAULT_STATUS_FORCELIST,
+        backoff_factor: Optional[float] = http_session_utils.DEFAULT_BACKOFF_FACTOR
+        ) -> Dict[str, Any]:
         """
         Makes a prediction using input `x`.
 
-        .. versionadded:: 0.13.14
-           The `always_retry_404` parameter.
-        .. versionadded:: 0.13.12
-           The `always_retry_429` parameter.
+        .. versionadded:: 0.22.0
+           The `prediction_id` parameter.
+        .. versionadded:: 0.22.0
+           The `retry_status` parameter.
+        .. versionadded:: 0.22.0
+           The `backoff_factor` parameter.
```

Comment:
I'd recommend adding `.. deprecated` directives for the `always_retry_*` parameters, unless there's a particular reason not to.

Reply:
Doc strings updated